crates/kitsune_p2p/direct/src/v1.rs

```diff
@@ -860,26 +847,26 @@ async fn handle_call(
 }
 
 async fn handle_gossip(
-    kdirect: Arc<Kd1>,
-    space: Arc<KitsuneSpace>,
-    to_agent: Arc<KitsuneAgent>,
-    ops: Vec<(Arc<KitsuneOpHash>, Vec<u8>)>,
+    _kdirect: Arc<Kd1>,
+    _space: Arc<KitsuneSpace>,
+    _ops: Vec<(Arc<KitsuneOpHash>, Vec<u8>)>,
 ) -> KdResult<()> {
-    for (op_hash, op_data) in ops {
-        let entry = KdEntrySigned::from_wire(op_data.into_boxed_slice())
-            .await
-            .map_err(KdError::other)?;
-        let op_hash = KdHash::from_kitsune_op_hash(&op_hash);
-        if &op_hash != entry.hash() {
-            return Err("data did not hash to given hash".into());
-        }
-        let root = KdHash::from_kitsune_space(&space);
-        let to_agent = KdHash::from_kitsune_agent(&to_agent);
-
-        kdirect.persist.store_entry(root, to_agent, entry).await?;
-    }
-
-    Ok(())
+    // for (op_hash, op_data) in ops {
+    //     let entry = KdEntrySigned::from_wire(op_data.into_boxed_slice())
+    //         .await
+    //         .map_err(KdError::other)?;
+    //     let op_hash = KdHash::from_kitsune_op_hash(&op_hash);
+    //     if &op_hash != entry.hash() {
+    //         return Err("data did not hash to given hash".into());
+    //     }
+    //     let root = KdHash::from_kitsune_space(&space);
+    //     let to_agent = KdHash::from_kitsune_agent(&to_agent);
+
+    //     kdirect.persist.store_entry(root, to_agent, entry).await?;
+    // }
+
+    // Ok(())
+    todo!()
```

Comment:
@neonphog I'm not sure how to make this work without a `to_agent`. To agent doesn't really make sense once we are gossiping between nodes. Any idea how I can change this to not need it?

Reply:
direct is lagging behind the holochain updates. It is full sync, and has separate stores per cell even for the same space... At this point I'm starting to wonder if it's worth maintaining. Are we actually getting anything out of it?

If we do want to fix it, for now you would probably need to iterate over all the local agents in that space an apply this operation on them all. It will lead to some duplication of work, because a higher level is already publishing to everyone on that node : )